packages/aztec.js/src/abiEncoder/inputCoder.js

```diff
@@ -0,0 +1,70 @@
+
+const { padLeft } = require('web3-utils');
+
+const encoderFactory = require('./encoderFactory');
+
+const inputCoder = {};
+
+
+inputCoder.bilateralSwap = (proofData, challenge, inputOwners, outputOwners, metadata) => {
+    const configs = {
+        CHALLENGE: challenge.slice(2),
```

Comment:
We should extend this to create a typing system for our zero-knowledge proof inputs, I'll write up a spec that describes this